homeassistant/components/telegram_bot/webhooks.py

```diff
@@ -108,19 +127,23 @@ class PushBotView(HomeAssistantView):
     url = TELEGRAM_WEBHOOK_URL
     name = "telegram_webhooks"
 
-    def __init__(self, hass, bot, dispatcher, trusted_networks):
+    def __init__(self, hass, bot, dispatcher, trusted_networks, secret_token):
         """Initialize by storing stuff needed for setting up our webhook endpoint."""
         self.hass = hass
         self.bot = bot
         self.dispatcher = dispatcher
         self.trusted_networks = trusted_networks
+        self.secret_token = secret_token
 
     async def post(self, request):
         """Accept the POST from telegram."""
         real_ip = ip_address(request.remote)
         if not any(real_ip in net for net in self.trusted_networks):
             _LOGGER.warning("Access denied from %s", real_ip)
             return self.json_message("Access denied", HTTPStatus.UNAUTHORIZED)
+        if self.secret_token != request.headers["X-Telegram-Bot-Api-Secret-Token"]:
```

Comment:
If the incoming request does not have the `"X-Telegram-Bot-Api-Secret-Token"` header, this will raise a KeyError. Let's check for that 👍